keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/Hamiltonian.java

```diff
@@ -71,7 +74,24 @@ public static Hamiltonian withDefaultConfig(KeanuRandom random) {
     public NetworkSamples getPosteriorSamples(final BayesianNetwork bayesNet,
                                               final List<? extends Vertex> fromVertices,
                                               final int sampleCount) {
+        return generatePosteriorSamples(bayesNet, fromVertices)
+            .generate(sampleCount);
+    }
+
+    public NetworkSamples getPosteriorSamples(final BayesianNetwork bayesNet,
+                                              final Vertex fromVertex,
+                                              final int sampleCount) {
+        return getPosteriorSamples(bayesNet, Collections.singletonList(fromVertex), sampleCount);
+    }
+
+    public NetworkSamplesGenerator generatePosteriorSamples(final BayesianNetwork bayesNet,
```

Comment:
Maybe we should be calling this something like `getPosteriorSamplesGenerator`. I don't think I like the `generatePosteriorSamples().generate()` chain